stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/state/channel/info/ChannelInfoEvent.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2014-2025 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.ui.common.state.channel.info
+
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+
+/**
+ * Represents events related to channel information actions, such as renaming, muting, hiding, or deleting a channel.
+ *
+ * This sealed interface defines various event types that can occur during channel operations,
+ * including success and error events for specific actions.
+ */
+@ExperimentalStreamChatApi
+public sealed interface ChannelInfoEvent {
+    /**
+     * Indicates an error occurred while renaming a channel.
+     *
+     * @param message The error message.
+     */
+    public data class RenameChannelError(val message: String) : ChannelInfoEvent
```

Comment:
Instead of a "error message", can't we provide more useful data here?
I assume this error is because some error on the API/Network/Whatever, and we should be able to provide a `Result.Failure` or something similar that help to understand why it failed